core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -0,0 +1,212 @@
+@SuppressLint("ViewConstructor")
+@SuppressWarnings("LongParameterList")
+open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
+  context: Context,
+  callback: WebViewCallback,
+  attrs: AttributeSet?,
+  nonVideoView: ViewGroup?,
+  videoView: ViewGroup?,
+  webViewClient: CoreWebViewClient?
+) : VideoEnabledWebView(context, attrs) {
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  private val callback: WebViewCallback
+
+  private val compositeDisposable = CompositeDisposable()
+
+  private fun setWindowVisibility(systemUiVisibility: Int) {
+    (context as Activity).window.decorView.systemUiVisibility = systemUiVisibility
+  }
+
+  init {
+    if (BuildConfig.DEBUG) {
+      WebView.setWebContentsDebuggingEnabled(true)
+    }
+    this.callback = callback
+    coreComponent.inject(this)
+    // Set the user agent to the current locale so it can be read with navigator.userAgent
+    val settings = settings
+    settings.userAgentString = "${getCurrentLocale(context)}"
+    settings.domStorageEnabled = true
+    settings.javaScriptEnabled = true
+    settings.loadWithOverviewMode = true
+    settings.useWideViewPort = true
+    setInitialScale(INITIAL_SCALE)
+    settings.builtInZoomControls = true
+    settings.displayZoomControls = false
+    clearCache(true)
+    settings.allowUniversalAccessFromFileURLs = true
+    setWebViewClient(webViewClient)
+    val client = KiwixWebChromeClient(callback, nonVideoView, videoView, this)
+    client.setOnToggledFullscreen { fullscreen ->
+      setWindowVisibility(
+        if (fullscreen) View.SYSTEM_UI_FLAG_LOW_PROFILE else View.SYSTEM_UI_FLAG_VISIBLE
+      )
+    }
+    webChromeClient = client
+  }
+
+  override fun performLongClick(): Boolean {
+    val result = hitTestResult
+    if (result.type == HitTestResult.SRC_ANCHOR_TYPE) {
+      callback.webViewLongClick(result.extra)
+      return true
+    }
+    return super.performLongClick()
+  }
+
+  override fun onCreateContextMenu(menu: ContextMenu) {
+    super.onCreateContextMenu(menu)
+    val result = hitTestResult
+    if (result.type == HitTestResult.IMAGE_ANCHOR_TYPE ||
+      result.type == HitTestResult.IMAGE_TYPE ||
+      result.type == HitTestResult.SRC_IMAGE_ANCHOR_TYPE
+    ) {
+      val saveMenu =
+        menu.add(0, 1, 0, resources.getString(R.string.save_media))
+      saveMenu.setOnMenuItemClickListener {
+        val msg = SaveHandler(zimReaderContainer).obtainMessage()
+        requestFocusNodeHref(msg)
+        true
+      }
+    }
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    compositeDisposable.add(
+      sharedPreferenceUtil.textZooms
+        .subscribe {
+          settings.textZoom = it
+        }
+    )
+  }
+
+  override fun onDetachedFromWindow() {
+    super.onDetachedFromWindow()
+    compositeDisposable.clear()
+  }
+
+  override fun onScrollChanged(
```

Comment:
this method definition could probably all be 1 line

Reply:
done